docs/src/modules/components/ApiPage/sections/ToggleDisplayOption.tsx

```diff
@@ -63,32 +68,27 @@ const getOption = (storageKey: string) => {
   } catch (error) {
     return DEFAULT_LAYOUT;
   }
-};
+}
 
 export function useApiPageOption(
   storageKey: string,
 ): [ApiDisplayOptions, (newOption: ApiDisplayOptions) => void] {
   const [option, setOption] = React.useState(getOption(storageKey));
-  const [needsScroll, setNeedsScroll] = React.useState(false);
 
   useEnhancedEffect(() => {
     neverHydrated = false;
     const newOption = getOption(storageKey);
     setOption(newOption);
-    setNeedsScroll(newOption !== DEFAULT_LAYOUT);
   }, [storageKey]);
 
   React.useEffect(() => {
-    setNeedsScroll(false);
-    if (needsScroll) {
-      return () => {
-        const id = document?.location.hash.slice(1);
-        const element = document.getElementById(id);
-        element?.scrollIntoView();
-      };
+    if (option !== DEFAULT_LAYOUT) {
+      const id = document.location.hash.slice(1);
+      const element = document.getElementById(id);
+      element?.scrollIntoView();
     }
-    return () => {};
-  }, [needsScroll]);
```

Comment:
I shouldn't have changed this, introducing the logic back in #40395. It led to a bug, the scroll override after first render (the effect re-run).